test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -6252,7 +6253,35 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
         rc = None
 
         if isinstance(obj, Job):
-            if script is None and obj.script is not None:
+            if self.platform == 'cray' or self.platform == 'craysim':
+                m = None
+                vncompute = False
+                if 'Resource_List.select' in obj.attributes:
+                    m = re.match('.*vntype=cray_compute.*',
+                                 obj.attributes['Resource_List.select'])
```

Comment:
Your re.match() is a little too permissive.  You will match any resource suffixed with vntype (e.g., foovntype).  Now you might be fine with this.  The correct regex will be hairy.  You need to check two things.  First if vntype is at the front of the line.  Second if it isn't at the front of the line, check if ':vntype' is in the line.  Since you're doing simple string matches you could get around this whole thing by breaking this into two string searches.  You can say m = obj.attributes['Resource_List.select'].startswith('vntype=cray_compute') or ':vntype=cray_compute' in obj.attributes['Resource_List.select']